src/pages/playground/Playground.tsx

```diff
@@ -251,14 +253,14 @@ export async function handleHash(
     // By default, use the first editor tab.
     const activeEditorTabIndex = convertParamToInt(qs.tabIdx) ?? 0;
     dispatch(updateActiveEditorTabIndex(workspaceLocation, activeEditorTabIndex));
-
-    const variant: Variant =
-      sourceLanguages.find(
-        language => language.chapter === chapter && language.variant === qs.variant
-      )?.variant ?? Variant.DEFAULT;
-
     if (chapter) {
-      props.handleChapterSelect(chapter, variant);
+      // TODO: To migrate the state logic away from playgroundSourceChapter
+      //       and playgroundSourceVariant into the language config instead
+      const languageConfig = getLanguageConfig(chapter, qs.variant as Variant);
```

Comment:
If this is temporary, I won't block on this. Otherwise, I'm not comfortable with this type cast since `qs.variant` can be some garbage value.

Reply:
1. It is temporary
2. We are using `getLanguageConfig` which will catch the garbage value and throw an error (as it would result in not being able to find a language config that matches the constraints), so even now it's still safe: https://github.com/source-academy/frontend/blob/93396dc689e743e5f517e813a93e7ffe520c8b6d/src/commons/application/ApplicationTypes.ts#L280-L292

Reply:
Hmm, the type cast is a code smell. Should the type of `variant` be made wider in `getLanguageConfig` then if it accepts and handles garbage values correctly? Can ignore if the comment on line 280 means that this function is just temporary.